# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1330136195 28800
#      Fri Feb 24 18:16:35 2012 -0800
# Node ID 98cf8aa9f8ba3a6cfe0660e517a4e5fe84d91c37
# Parent  7d5a653825b9450fb33ede6f8475a0a46141789a
prepad.m: Improve input validation messages and tests.

* prepad.m: Improve input validation messages and tests.

diff --git a/scripts/general/prepad.m b/scripts/general/prepad.m
--- a/scripts/general/prepad.m
+++ b/scripts/general/prepad.m
@@ -41,34 +41,34 @@ function y = prepad (x, l, c, dim)
   if (nargin < 2 || nargin > 4)
     print_usage ();
   endif
 
   if (nargin < 3 || isempty (c))
     c = 0;
   else
     if (! isscalar (c))
-      error ("prepad: third argument must be empty or a scalar");
+      error ("prepad: pad value C must be empty or a scalar");
     endif
   endif
 
   nd = ndims (x);
   sz = size (x);
   if (nargin < 4)
     ## Find the first non-singleton dimension.
     (dim = find (sz > 1, 1)) || (dim = 1);
   else
     if (!(isscalar (dim) && dim == fix (dim))
         || !(1 <= dim && dim <= nd))
       error ("prepad: DIM must be an integer and a valid dimension");
     endif
   endif
 
   if (! isscalar (l) || l < 0)
-    error ("prepad: second argument must be a positive scaler");
+    error ("prepad: length L must be a positive scalar");
   endif
 
   if (dim > nd)
     sz(nd+1:dim) = 1;
   endif
 
   d = sz (dim);
 
@@ -92,10 +92,15 @@ endfunction
 
 %!assert (prepad ([1,2], 2, 2, 1), [2,2;1,2])
 
 ## FIXME -- we need tests for multidimensional arrays.
 
 %!error prepad ()
 %!error prepad (1)
 %!error prepad (1,2,3,4,5)
-%!error prepad ([1,2], 2, 2,3)
+%!error <C must be empty or a scalar> prepad ([1,2], 2, ones (2))
+%!error <DIM must be an integer> prepad ([1,2], 2, 2, ones (3))
+%!error <DIM must be an integer> prepad ([1,2], 2, 2, 1.1)
+%!error <DIM must be an integer> prepad ([1,2], 2, 2, 3)
+%!error <L must be a positive scalar> prepad ([1,2], ones (2))
+%!error <L must be a positive scalar> prepad ([1,2], -1)
 
